### PR TITLE
[Hexagon] Add E2E test demonstrating how to apply blocked layout schedule to conv2d via metaschedule

### PR DESCRIPTION
**Pull request by csullivan**
Demonstrates the use of a custom search space generation from a manual written schedule. In this there are no degrees of freedom so tuning will not improve the schedule currently, but the test demonstrates the schedule-time layout transformation of the conv2d primfunc to the 8x8x32 packed layout for convolution ops.

### CODE REVIEW

**Comment by masahi** [comment]
@tvm-bot rerun

File: tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py
Comment (masahi) [inline tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:372]: Do we want perf measurement and detailed profiling report from this test?

Comment (csullivan) [reply tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py]: Thank you and removed. This is indeed unnecessary for this test.

File: tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py
Comment (masahi) [inline tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:285]: cc @zxybazh so by using `ScheduleFn` here, MS won't be looking at non conv2d blocks?

**Review by masahi — APPROVED** [review]
LGTM modulo the confirmation on TODO

File: tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py
Comment (zxybazh) [reply tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:285]: By using `ScheduleFn` here, the search space is solely generated by the schedule function over the whole workload. In this case, if you're using `_schedule_packed_8x8x32_conv2d`'s `schedule_fn` to construct the `ScheduleFn` space generator, it will only look at conv2d blocks because the function doesn't transform any of the non-conv2d blocks right?

Comment (masahi) [reply tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:285]: Yeah that's my understanding.

Comment (masahi) [reply tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:285]: @csullivan so shall we remove the TODO comment (which also has a typo lol)

Comment (csullivan) [reply tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:285]: :) Yes, but one more question, if one utilizes a `ScheduleFn` to define a search space, but also sets custom `sch_rules` to be applied per block as, 
```
            space=ms.space_generator.ScheduleFn(
                schedule_conv2d_for_tune,
                sch_rules=[some set of rules] , # <<<<
                postprocs=[],
                mutator_probs={},
            ),
```
what will be the interaction between the scheduling function and the rules?

Comment (zxybazh) [reply tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:285]: For `ScheduleFn` I don't think there will be any interaction between the scheduling function and the rules, they will only be used for `PostOrderApply` space generator.

Comment (masahi) [reply tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:285]: @csullivan can you remove the TODO comment?

File: tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py
Comment (masahi) [inline tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py:200]: My PR https://github.com/apache/tvm/pull/13206 has `has_block(sch, block_name)` utility, which can remove this hack. I'll update this code after it is merged.

**Comment by csullivan** [comment]
Thanks for patience, let us hold off on merging this one for a bit as there are a few issues that arise at the end of a tuning run with this test:
1. Lack of JSON serialization for tir::IndexMap in src/meta_schedule/database/database_utils.cc::JSONDumps
2. Metascheduler search workers hang when the search space has only one sample.

**Comment by zxybazh** [comment]
Hi Chris, regarding (2) does the worker hang when the search space that has only one sample, and will not terminate by itself even after 5 iterations?

**Comment by csullivan** [comment]
Thanks @zxybazh for following the thread closely. Looks like (2) was a red herring and is not reproducible for me any longer. Furthermore, (1) is coming in a follow up PR and this test is skipped so no harm merging this now.

**Pull request merged**
